### ISSUE TITLE
 free(): invalid next size (fast) 

### ISSUE THREAD

**Issue by feihongloveworld** [CCCKW/mHapTk#1]
when i identity the MHB on WGBS data within a 1M region, i could not get right result and have a core dump;
“”“
*** Error in `/miniconda3/envs/py37/bin/python': free(): invalid next size (fast): 0x0000000001cb2540 ***
”“”

**Comment by JiantaoShi** [comment]
The python code takes all mHaps in the predefined regions for downstream analysis. In the published study, we actually segmented the genome for MHB discovery. You can try our Java implementation mHapSuite, which is extensively optimized for MHB discovery:
https://github.com/yoyoong/mHapSuite

**Comment by feihongloveworld** [comment]
> 

thank you for your wonderful job, i've read all the python code and leant a lot from it.
and i'll try this mHapSuit you recommend.

another question is about mhaptools that is released by your lab. Could mhaptools covert bams generated by bwa-meth ? 
waiting for your reply
wish you have a nice day.

**Comment by JiantaoShi** [comment]
Yes, please try mHapSuit. If you run into memory issues, you can try the genome-wide mode for MHB discovery (just don't specify -region and -bedPath options). In this mode, mHapSutie will read mHaps chunk by chunk, so it should be no memory issues. 

Regarding bwa-meth, we believe it should work. The convert function in mHapSutie can do this as well. Let us know if you find any issues.

**Comment by feihongloveworld** [comment]
> Yes, please try mHapSuit. If you run into memory issues, you can try the genome-wide mode for MHB discovery (just don't specify -region and -bedPath options). In this mode, mHapSutie will read mHaps chunk by chunk, so it should be no memory issues.
> 
> Regarding bwa-meth, we believe it should work. The convert function in mHapSutie can do this as well. Let us know if you find any issues.

thank you very much !
